app/safecast/tasks/device_story/update_metadata.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+module Tasks
+  module DeviceStory
+    class UpdateMetadata
+      class << self
+        def call
+          new.call
+        end
+      end
+
+      def call
+        each_metadata(parse(fetch(devices_uri))) do |device_urn, metadata|
+          begin
+            find_or_update_device(device_urn, metadata)
+          rescue ActiveRecord::RecordInvalid
+            warn "Could not update device metadata for #{device_urn}."
+          end
+        end
+      end
+
+      private
+
+      def find_or_update_device(device_urn, metadata)
+        device = ::DeviceStory::Device.find_by(device_urn: device_urn)
+        if device
+          device.update!(metadata: metadata)
+        else
+          ::DeviceStory::Device.create!(device_urn: device_urn, metadata: metadata)
+        end
+      end
+
+      def each_metadata(metadatas)
+        metadatas.each do |metadata|
+          device_urn = metadata['device_urn']
+          unless device_urn
+            warn 'Could not find device_urn.'
+            next
+          end
+          yield device_urn, metadata
+        end
+      end
+
+      def parse(json)
+        JSON.parse(json)
+      rescue Psych::SyntaxError
```

Comment:
This may be a difference in programming style, but this seems to quietly hide an exception. Is there any harm in allowing the exception to propagate? I feel like we might also get better debugging information that way -- if `SyntaxError` contains any useful information, it is lost here.

I may be thinking about this in terms of Java unchecked exceptions.